src/patternfly/components/DataList/data-list-grid.scss

```diff
@@ -0,0 +1,78 @@
+@mixin pf-c-data-list-responsive-block() {
+  // base responsive styles
+  @at-root .pf-c-data-list:not([class*="pf-m-grid"]) {
+    @media screen and (min-width: $pf-global--breakpoint--md) {
+      @content;
+    }
+  }
+
+  @at-root .pf-c-data-list.pf-m-grid-none {
+    @media screen and (min-width: $pf-global--breakpoint--xs) {
+      @content;
+    }
+  }
+
+  // grid modifiers
+  @each $size, $bp in $pf-global--breakpoint-name-map {
+    @at-root .pf-c-data-list.pf-m-grid-#{$size} {
+      @media screen and (min-width: #{$bp}) {
+        @content;
+      }
+    }
+  }
+}
+
+@include pf-c-data-list-responsive-block {
+  --pf-c-data-list__cell--cell--PaddingTop: var(--pf-c-data-list__cell--cell--md--PaddingTop);
```

Comment:
Can you make a note here to rethink the variable naming during breaking change period? Maybe to something like `--flex--` or `--desktop--`. Otherwise, very nice!